src/parser/TripleComponent.cpp

```diff
@@ -50,7 +50,9 @@ std::optional<Id> TripleComponent::toValueIdIfNotString() const {
       return std::nullopt;
     } else if constexpr (std::is_same_v<T, Literal>) {
       auto geopoint = GeoPoint::parseFromLiteral(value);
-      if (geopoint.has_value()) return Id::makeFromGeoPoint(geopoint.value());
+      if (geopoint.has_value()) {
+        return Id::makeFromGeoPoint(geopoint.value());
+      }
```

Comment:
I know that I initially told you to put this implementation here,
but now I remember that this was just to initially get something working.
You should move this logic to
the function `literalAndDatatypeToTripleComponentImpl` in the `TurtleParser` class in the `RdfParser.cpp` file (line 460ff).

There we have all the other logic for folding types etc. Then we have one centralized place, that ensures that we don't miss anything anywhere.